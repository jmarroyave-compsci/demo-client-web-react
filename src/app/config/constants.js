import _package_ from 'package.json';

const Constants = {
	DEBUG : process.env.NEXT_PUBLIC__DEBUG,
	LOCAL : process.env.NEXT_PUBLIC__LOCAL,
	APP : {
		BASE_PATH: process.env.NEXT_PUBLIC__SITE_URL,
		SITE_URL: process.env.NEXT_PUBLIC__SITE_URL,
		VERSION : _package_.version,
		TITLE: process.env.NEXT_PUBLIC__TITLE,
		PAGE_TITLE: process.env.NEXT_PUBLIC__PAGE_TITLE
	},
	SERVERS: {
		DATA: process.env.NEXT_PUBLIC__DATA, 
		GRAPHQL :  process.env.NEXT_PUBLIC__DATA_GRAPHQL,
	},
	DEFAULTS : {
		PAGING_PAGE_SIZE: 10,
	},
	PLUGINS : {
		GOOGLE: {
			ANALYTICS: process.env.NEXT_PUBLIC_GOOGLE_ANALYTICS,
			MAPS: process.env.NEXT_PUBLIC__GOOGLE_MAPS_KEY,
		},
	},
	FEATURES : {
		SEARCH : (process.env.NEXT_PUBLIC__FEATURE_SEARCH == "true"),
	},
}

export default Constants;