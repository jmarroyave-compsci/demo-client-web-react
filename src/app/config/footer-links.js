import _links from './links';

const links = [
  { "name" : "jmarroyave", items: [
    { "name": "blog", "link" : _links.blog },
  ] },
  { "name" : "labs", items: [
    { "name": "stackoverflow", "link" : _links.labs_trends },
    { "name": "mlb", "link" : _links.labs_mlb },
    { "name": "search", "link" : _links.labs_search },
  ] },
  { "name" : "knowledge base", items: [
    { "name": "android", "link" : _links.kb_android },
  ] },
];

export default links;