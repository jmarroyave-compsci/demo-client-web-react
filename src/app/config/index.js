import _package_ from 'package.json';

const Constants = {
	TEST: (process.env.TESTING === "true") ? true : false,
	DEBUG : process.env.NEXT_PUBLIC__DEBUG,
	LOCAL : process.env.NEXT_PUBLIC__LOCAL,
	APP : {
		BASE_PATH: process.env.NEXT_PUBLIC__BASE_PATH,
		SITE_URL: process.env.NEXT_PUBLIC__SITE_URL,
		VERSION : _package_.version,
		TITLE: process.env.NEXT_PUBLIC__TITLE,
		DESCRIPTION: process.env.NEXT_PUBLIC__DESCRIPTION,
		PAGE_TITLE: process.env.NEXT_PUBLIC__PAGE_TITLE,
		HOME_URL: `${process.env.NEXT_PUBLIC__SITE_URL}${process.env.NEXT_PUBLIC__BASE_PATH}`,
	},
	SERVERS: {
		DATA: process.env.NEXT_PUBLIC__SERVER, 
		WEB_SOCKETS: process.env.NEXT_PUBLIC__SERVER, 
		GRAPHQL :  `${process.env.NEXT_PUBLIC__SERVER}/${process.env.NEXT_PUBLIC__SERVER_API_VERSION}/graphql`,
	},
	DEFAULTS : {
		PAGING_PAGE_SIZE: 10,
		IMAGE_PLACEHOLDER: "/img/ph.jpg",
	},
	PLUGINS : {
		GOOGLE: {
			ANALYTICS: process.env.NEXT_PUBLIC_GOOGLE_ANALYTICS,
			MAPS: process.env.NEXT_PUBLIC__GOOGLE_MAPS_KEY,
		},
	},
	FEATURES : {
		SEARCH : (process.env.NEXT_PUBLIC__FEATURE_SEARCH == "true"),
	},
	SOCIAL_NETWORKS: {
		CARD : {
			TITLE : `${process.env.NEXT_PUBLIC__TITLE} v.${_package_.version}`,
			THUMBNAIL_TWITTER: `${process.env.NEXT_PUBLIC__BASE_PATH}${process.env.NEXT_PUBLIC__CARD_THUMBNAIL_TWITTER}`,
		},
		TWITTER: process.env.NEXT_PUBLIC__SOCIAL_NETWORKS_TWITTER,
	}
}

export default Constants;