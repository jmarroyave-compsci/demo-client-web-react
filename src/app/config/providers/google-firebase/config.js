const config = {
  apiKey: "AIzaSyDaNheMMUtsqx8AjiJAWDcfXwAEBYLJUVY",
  authDomain: "labs-f885b.firebaseapp.com",
  projectId: "labs-f885b",
  //projectId: "labs-f885",
  storageBucket: "labs-f885b.appspot.com",
  messagingSenderId: "606766444808",
  appId: "1:606766444808:web:41eec12c129d582727bcbd",
  measurementId: "G-TNBYCW57TS"
};

export default config;