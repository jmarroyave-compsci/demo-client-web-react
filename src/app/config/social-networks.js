const links = [
  /*
	{enable: true, url: 'https://plus.google.com/112618720945759819017', icon: 'google-plus', color: '#d34836'},
	{enable: true, url: 'https://twitter.com/jmarroyave', icon: 'twitter', color: '#00aced'},
	*/
  { enable: true, url: 'https://facebook.com/jmarroyave', icon: 'facebook', color: '#3b5998' },
  { enable: true, url: 'https://www.linkedin.com/in/jmarroyave', icon: 'linkedin', color: '#0077B5' },
  { enable: true, url: 'https://github.com/jmarroyave', icon: 'github', color: '#000' },
  { enable: true, url: 'mailto:jmarroyave.compsci@gmail.com', icon: 'envelope-o', color: '#000' },
];


export default links;