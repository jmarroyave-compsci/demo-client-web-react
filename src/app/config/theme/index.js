import React from 'react';
import { createTheme } from '@mui/material/styles';
import { red } from '@mui/material/colors';

// Create a theme instance.
export const getTheme = ( mode ) => {
  return createTheme({
    palette: {
      mode: mode,
      primary: {
        main: '#556cd6',
      },
      secondary: {
        main: '#19857b',
      },
      error: {
        main: red.A400,
      },
    },
  });
}