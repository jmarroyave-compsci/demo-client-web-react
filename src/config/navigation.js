
class Navigation {
  constructor(){
    this.drawer = {
      show: false,
    };

    this.options = [
      {
        showInMenu: true, 
        showInDrawer: true, 
        name: 'Home', 
        icon: 'home', 
        route: '/', 
      },
      {
        showInMenu: true, 
        showInDrawer: true, 
        separator: true,
      },
      {
        showInMenu: true, 
        showInDrawer: true, 
        name: 'Movies', 
        icon: 'movie', 
        route: '/movies', 
      },
        {
          showInMenu: false, 
          showInDrawer: false, 
          name: 'Movie', 
          icon: 'movie', 
          route: '/movie/:id/:name', 
        },
      {
        showInMenu: true, 
        showInDrawer: true, 
        name: 'Podcasts', 
        icon: 'podcasts', 
        route: '/podcasts', 
      },
        {
          showInMenu: false, 
          showInDrawer: false, 
          name: 'Podcast', 
          icon: 'podcast', 
          route: '/podcast/:id/:name', 
        },
      {
        showInMenu: true, 
        showInDrawer: true, 
        separator: true,
      },
      {
        showInMenu: true, 
        showInDrawer: true, 
        name: 'People', 
        icon: 'people', 
        route: '/people', 
      },
        {
          showInMenu: false, 
          showInDrawer: false, 
          name: 'Person', 
          icon: 'person', 
          route: '/person/:id/:name', 
        },
      {
        showInMenu: false, 
        showInDrawer: false, 
        name: 'Awards', 
        route: '/stories/awards/:entity?/:year?/:page?', 
      },
      {
        showInMenu: false, 
        showInDrawer: false, 
        name: 'Stories', 
        route: '/stories', 
      },
      {
        showInMenu: false, 
        showInDrawer: false, 
        name: 'PodcastMusic', 
        route: '/podcasts/stories/music/:page?', 
      },
      {
        showInMenu: true, 
        showInDrawer: true, 
        separator: true,
      },
      {
        showInMenu: true, 
        showInDrawer: true, 
        name: 'About', 
        icon: 'info', 
        route: '/about', 
      },
      {
        showInMenu: false, 
        showInDrawer: false, 
        name: 'Search', 
        icon: 'search', 
        route: '/search/:qry/:page?', 
      },
    ];
  }
}



const nav = new Navigation();
export default nav;