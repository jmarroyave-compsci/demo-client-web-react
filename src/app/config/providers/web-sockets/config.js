import config from 'app/config';

const CONF = {
  SERVER: config.SERVERS.WEB_SOCKETS
};

export default CONF;