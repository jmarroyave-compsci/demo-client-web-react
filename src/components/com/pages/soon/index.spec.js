import React from 'react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import {render, fireEvent, waitFor, screen} from '@testing-library/react'
import '@testing-library/jest-dom'

import { itMustHaveNoErrors } from "lib/tests/shared"

import Page from '.';
import config from './.config';

const mockStore = configureStore([]);
 
describe('Search Page', () => {
  let store;
  let component;
 
  beforeEach(() => { 
    var state = {}
    store = mockStore(state);
    component = (
      <Provider store={store}>
        <Page />
      </Provider>
    );    
	});
 
 
  it('compare against snapshot', () => {
    const { container } = render(component)
    itMustHaveNoErrors(container)
    expect(container).toMatchSnapshot();
  });
});
