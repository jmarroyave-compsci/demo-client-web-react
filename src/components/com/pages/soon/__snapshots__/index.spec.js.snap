// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Page compare against snapshot 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  font-size: 1rem;
}

@media (min-width:0px) {
  .emotion-1 {
    padding: 0rem 2rem 3rem 2rem;
  }
}

@media (min-width:600px) {
  .emotion-1 {
    padding: 0rem 2rem 3rem 2rem;
  }
}

@media (min-width:1200px) {
  .emotion-1 {
    padding: 0rem 4rem 3rem 4rem;
  }
}

.emotion-2 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  width: 100%;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.emotion-3 {
  box-sizing: border-box;
  margin: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-basis: 100%;
  -ms-flex-preferred-size: 100%;
  flex-basis: 100%;
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  max-width: 100%;
}

@media (min-width:600px) {
  .emotion-3 {
    -webkit-flex-basis: 100%;
    -ms-flex-preferred-size: 100%;
    flex-basis: 100%;
    -webkit-box-flex: 0;
    -webkit-flex-grow: 0;
    -ms-flex-positive: 0;
    flex-grow: 0;
    max-width: 100%;
  }
}

@media (min-width:900px) {
  .emotion-3 {
    -webkit-flex-basis: 75%;
    -ms-flex-preferred-size: 75%;
    flex-basis: 75%;
    -webkit-box-flex: 0;
    -webkit-flex-grow: 0;
    -ms-flex-positive: 0;
    flex-grow: 0;
    max-width: 75%;
  }
}

.emotion-4 {
  padding: 0rem 0rem 4rem 0rem;
}

.emotion-5 {
  box-sizing: border-box;
  margin: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-basis: 100%;
  -ms-flex-preferred-size: 100%;
  flex-basis: 100%;
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  max-width: 100%;
}

@media (min-width:600px) {
  .emotion-5 {
    -webkit-flex-basis: 100%;
    -ms-flex-preferred-size: 100%;
    flex-basis: 100%;
    -webkit-box-flex: 0;
    -webkit-flex-grow: 0;
    -ms-flex-positive: 0;
    flex-grow: 0;
    max-width: 100%;
  }
}

@media (min-width:900px) {
  .emotion-5 {
    -webkit-flex-basis: 25%;
    -ms-flex-preferred-size: 25%;
    flex-basis: 25%;
    -webkit-box-flex: 0;
    -webkit-flex-grow: 0;
    -ms-flex-positive: 0;
    flex-grow: 0;
    max-width: 25%;
  }
}

.emotion-6 {
  background-color: #fff;
  color: rgba(0, 0, 0, 0.87);
  -webkit-transition: box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  border-radius: 4px;
  box-shadow: 0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12);
}

.emotion-7 {
  color: rgba(0, 0, 0, 0.87);
  background-color: #fff;
  padding: 1rem;
}

.emotion-8 {
  font-size: 1.5rem;
  padding-bottom: 0.5rem;
  color: inherit;
}

.emotion-9 {
  font-size: 0.9rem;
  color: inherit;
}

.emotion-10 {
  cursor: pointer;
  color: inherit;
  background-color: transparent;
  -webkit-text-decoration: underline;
  text-decoration: underline;
  text-decoration-style: dotted;
  text-decoration-color: inherit;
  text-decoration-thickness: 1px;
  text-overflow: ellipsis;
  white-space: nowrap;
  overflow: hidden;
  display: inline-block;
  width: 100%;
}

<div>
  <div
    class="emotion-0"
  >
    <div
      class="emotion-1"
    >
      <div
        class="MuiGrid-root MuiGrid-container page-module emotion-2"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12 MuiGrid-grid-md-9 emotion-3"
        >
          <div
            class="emotion-4"
          >
            <div>
              SOOOOOON
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12 MuiGrid-grid-md-3 emotion-5"
        >
          <div
            style="padding-left: 1rem; padding-right: 0.5rem;"
          >
            <div
              class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 emotion-6"
            >
              <div
                class="emotion-7"
              >
                <div
                  class="emotion-8"
                >
                  See also
                </div>
                <div
                  class="emotion-9"
                >
                  <a
                    href="/about/log"
                    style="flex: 1; display: flex; width: 100%;"
                  >
                    <span
                      class="emotion-10"
                    >
                      Project's log
                    </span>
                  </a>
                </div>
                <div
                  class="emotion-9"
                >
                  <a
                    href="/about/timeline"
                    style="flex: 1; display: flex; width: 100%;"
                  >
                    <span
                      class="emotion-10"
                    >
                      Project's timeline
                    </span>
                  </a>
                </div>
                <div
                  class="emotion-9"
                >
                  <a
                    href="/podcasts"
                    style="flex: 1; display: flex; width: 100%;"
                  >
                    <span
                      class="emotion-10"
                    >
                      Podcasts Dashboard
                    </span>
                  </a>
                </div>
                <div
                  class="emotion-9"
                >
                  <a
                    href="/movies/stories/topics"
                    style="flex: 1; display: flex; width: 100%;"
                  >
                    <span
                      class="emotion-10"
                    >
                      Movie topics
                    </span>
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
